test/06_device-log.ts

```diff
@@ -222,4 +221,53 @@ describe('device log', () => {
 			'streamed log line 1',
 		]);
 	});
+
+	it('should rate limit stream-read device', async () => {
+		const dummyLogs = [createLog({ message: `not rate limited` })];
+		await supertest(ctx.device.apiKey)
+			.post(`/device/v2/${ctx.device.uuid}/logs`)
+			.send(dummyLogs)
+			.expect(201);
+
+		async function testRatelimitedDeviceLogsStream() {
+			let evalValue;
+			const req = supertest(ctx.user)
+				.get(`/device/v2/${ctx.device.uuid}/logs`)
+				.query({
+					stream: 1,
+					count: 1,
+				})
+				.parse(function (res, callback) {
+					res.on('data', async function (chunk) {
+						const parsedChunk = JSON.parse(Buffer.from(chunk).toString());
+						evalValue = parsedChunk;
+						// if data stream provides proper data terminate the stream (abort)
+						if (typeof parsedChunk === 'object') {
+							if (parsedChunk?.message === 'not rate limited') {
+								req.abort();
+							}
+						}
```

Comment:
```suggestion
						if (typeof parsedChunk === 'object' && parsedChunk?.message === 'not rate limited') {
							req.abort();
						}
```
Nit. Also, arrays parse as `object` when tested with `typeof`, but this is just a test with known data shapes so probably don't need to worry about that too much.

Reply:
:+1: